ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/action/RevealResourceAction.java

```diff
@@ -74,13 +88,28 @@ public void actionPerformed(ActionEvent e) {
 
       checkState(!path.isEmpty());
 
+      ensureProjectExplorerPart();
       eventBus.fireEvent(new RevealResourceEvent(path));
     } else {
       final Resource[] resources = appContext.getResources();
 
       checkState(resources != null && resources.length == 1);
 
+      ensureProjectExplorerPart();
       eventBus.fireEvent(new RevealResourceEvent(resources[0]));
     }
   }
+
+  private void ensureProjectExplorerPart() {
+    PartStack navigationPartStack = workspaceAgent.getPartStack(NAVIGATION);
+    PartPresenter activePart = navigationPartStack.getActivePart();
+
+    if (!(activePart instanceof ProjectExplorerPresenter)) {
+      ProjectExplorerPresenter projectExplorerPresenter = projectExplorerPresenterProvider.get();
+      if (activePart == null) {
+        workspaceAgent.openPart(projectExplorerPresenter, NAVIGATION);
+      }
```

Comment:
Can we simplify this nested structure?
At first do the check for null and open project explorer in this case.
Next do the check 'instanceof' if active part is not null
WDYT?